doc/proposals/dualstack.md

```diff
@@ -0,0 +1,280 @@
+      ConfigurationPath   string           `json:"configuration_path"`
+      PodName             string
+      PodNamespace        string
+ }
+```
+
+Corresponding changes will also be required in `whereabouts/pkg/allocate/allocate.go`, `whereabouts/pkg/config/config.go` etc.
+
+<hr>
+
+## Alternative Design <a name="alternative"></a>
+
+Alternatively, if we only want to support DualStack (i.e. at max one IPv4 and one IPv6 address), we can have a simpler config change which is easily backward compatible. _(But I still prefer the first approach even though it is requires more effort)_
+
+We can have change in IPAM Config as below.
+
+<table>
+<tr>
+<th>Old IPAM Config</th>
+<th>Changes</th>
+<th>New IPAM Config</th>
+</tr>
+<tr>
+<td>
+  
+```json
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.2.225/28",
+        "exclude": [
+           "192.168.2.229/30",
+           "192.168.2.236/32"
+        ]
+      }
+}
+```
+  
+</td>
+<td>
+
+```diff
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.2.225/28",
++       "secondary_range": "2001::0/116",
+        "exclude": [
+           "192.168.2.229/30",
+           "192.168.2.236/32",
++          "2001::0/124"
+        ]
+      }
+}
+```
+
+</td>
+<td>
+
+```json
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.2.225/28",
+        "secondary_range": "2001::0/116",
+        "exclude": [
+           "192.168.2.229/30",
+           "192.168.2.236/32",
+           "2001::0/124"
+        ]
+      }
+}
+```
+
+</td>
+</tr>
+</table>
+
+And we can simply add `secondary_*` fields in `IPAMConfig`.
+
+<hr>
+
+### Summary <a name="summary"></a>
+
+Currently, we have above two approaches for supporting DualStack in whereabouts.
+
+First approach requires addition on a new _type_ for encapsulating all the IP related information on `IPAMConfig` and requires significant amount of change in the current code. But with this additional effort, it will have additional benefit of allowing us to allocate as many as we required IP addresses of any IP family without any constraints. _(We need to support the existing fields too for backward compatibility)_
+
+Second approach is somewhat simpler. It just adds some additional fields for _secondary_ IP address related configuration (which might be IPv4 or IPv6 depending on stack policy). The downside is that we will limit ourselves to adding at max 2 IP address for a pod. _(TBH, it also seems okay for now, but I prefer the first approach)_
```

Comment:
I agree with the analysis presented here: first approach is preferable - simply because the second one does **not** scale.

Furthermore, the IP is somewhat clunky IMO.

I vote for option 1 as well.